solutions/form-builder/src/app/app.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import {
+  Validators,
+  FormGroup,
+  FormControl,
+  FormBuilder
+} from '@angular/forms';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+
+  contacts = [];
+
+  firstName = new FormControl('', [
+    Validators.required,
+    Validators.minLength(2)
+  ]);
+  lastName = new FormControl('', [
+    Validators.required,
+    Validators.minLength(2)
+  ]);
+  email = new FormControl('', [
+    Validators.required,
+    isValidEmail
+  ]);
+
+  contactForm: FormGroup = this.builder.group({
+    firstName: this.firstName,
+    lastName: this.lastName,
+    email: this.email
+  });
+
+  constructor(private builder: FormBuilder) {}
+
+  addPerson() {
+    this.contacts.push(this.contactForm.value);
```

Comment:
I guess I didn't see this before. Can we use a non mutation version of this? Something like:

```ts
this.contacts = [...this.contacts, this.contactForm.value];
```